src/incidencehandler.cpp

```diff
@@ -101,30 +101,37 @@ bool IncidenceHandler::copiedPropertiesAreEqual(const KCalCore::Incidence::Ptr &
     RETURN_FALSE_IF_NOT_EQUAL(a, b, duration(), "duration");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, hasDuration(), "hasDuration");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, isReadOnly(), "isReadOnly");
-    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(a->dtStart() != b->dtStart(), "dtStart", (a->dtStart().toString() + " " + b->dtStart().toString()));
     RETURN_FALSE_IF_NOT_EQUAL(a, b, comments(), "comments");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, contacts(), "contacts");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, altDescription(), "altDescription");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, categories(), "categories");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, customStatus(), "customStatus");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, description(), "description");
-    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(!qFuzzyCompare(a->geoLatitude(), b->geoLatitude()), "geoLatitude", (QString("%1 %2").arg(a->geoLatitude()).arg(b->geoLatitude())));
-    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(!qFuzzyCompare(a->geoLongitude(), b->geoLongitude()), "geoLongitude", (QString("%1 %2").arg(a->geoLongitude()).arg(b->geoLongitude())));
+    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(!qFuzzyCompare(a->geoLatitude(), b->geoLatitude()), "geoLatitude", (QString("%1 != %2").arg(a->geoLatitude()).arg(b->geoLatitude())));
+    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(!qFuzzyCompare(a->geoLongitude(), b->geoLongitude()), "geoLongitude", (QString("%1 != %2").arg(a->geoLongitude()).arg(b->geoLongitude())));
     RETURN_FALSE_IF_NOT_EQUAL(a, b, hasGeo(), "hasGeo");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, location(), "location");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, secrecy(), "secrecy");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, status(), "status");
     RETURN_FALSE_IF_NOT_EQUAL(a, b, summary(), "summary");
 
+    // check recurrence information. Note that we only need to check the recurrence rules for equality if they both recur.
+    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(a->recurs() != b->recurs(), "recurs", a->recurs() + " != " + b->recurs());
+    RETURN_FALSE_IF_NOT_EQUAL_CUSTOM(a->recurs() && *(a->recurrence()) != *(b->recurrence()), "recurrence", "...");
```

Comment:
This should have no effect, but it turns out that if one has a valid-but-empty recurrence, and the other has invalid/null recurrence, the later comparison fails (even though they're equivalent, ie, no recurrence).  So, this is a workaround for that.